scripts/bulk-access-control.md

```diff
@@ -0,0 +1,93 @@
+
+## Parameters
+
+The script takes the following parameters:
+
+| Parameter                        | Description                                                                                                                                                                            |
+|----------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `-f`, <br>`--file`                | This parameter specifies the json file containing the processing instruction.                                                                                                              |
+| `-h`, <br>`--help`               | This parameter displays help information for the script.                                                                                                                               |
+
+The json file specified with the file parameter has the following structure
+ structured as follow
+```json
+{
+   target: [uuid1, uuid2, ..., uuidN],
+   item: {
+      mode: "replace",
+      accessConditions: [
+          {
+            "name": "openaccess"
+          }
+      ]
+   },
+   bitstream: {
+      constraints: {
+          uuid: [bit-uuid1, bit-uuid2, ..., bit-uuidN],
+      },
+      mode: "add",
+      accessConditions: [
+        {
+         "name": "embargo",
+         "startDate": "2024-06-24T23:59:59.999+0000"
+        },
+        {
+         "name": "network"         
+        }
+      ]
+   }
+}
+```
+* target: is a list of uuid of communities, collections and individual items to process
+* item: the item node contains the processing detail at the item level for the identified resources
+  * mode: can be replace or add. Replace will mean that after processing the items will have only the access conditions that are specified in the accessContitions property. Add will mean that after processing the items will have the specified accessCondition in addition to the ones already defined.
+  * accessConditions: the list of access conditions to apply, with their specific parameters if needed (such as a startDate for an embargo, etc.). Only in replace mode the accessConditions can be empty in this case the access conditions of the item will be reset to what is inherited from their owningCollection. Please note that when accessConditions is set and is not empty not inheritence from the owningCollection occur
+* bitstreams: the bitstream node contains the processing detail at the bitstream level for the identified resources
+  * constraints: when the target element contains just a single uuid related to an item it is possible to limit the bitstreams to process to a predefined list instead than process all the bitstream of the item. If the constraints node is not present, is empty or contains a null uuid or 0-size list all the bitstreams are processed
```

Comment:
What happens if the `target` is a list of UUIDs?  Is the `constraints` *ignored*?  I'm assuming so, but we should state that it's ignored if so.

It's also unclear to me if `constraints` is supposed to list the *bitstreams to be processed* or the *bitstreams to be ignored*.   I think it's the first one.  But, I'd recommend we improve the wording here and say something like this:
```
constraints: when the `target` element (see above) contains just a single Item UUID, then the 'constraints' can be used to limit which Bitstreams are processed in that Item. If `constraints` specifies one or more bitstream UUIDs, then only those bitstreams listed will be processed. If the `constraint`s node is not present, is empty or contains a null uuid or 0-size list, then all the bitstreams will be processed.  If the `target` element contains a Community/Collection UUID or a list of Item UUIDs, then any `constraints` values will be ignored and all the bitstreams will be processed.
```

Reply:
I have updated the wording. Please note that I have added a new section that clarify which rules are checked for the input file. If the file is invalid the script execution will fail, I would prefer that over ignoring elements that would imply error in the creation of the input file